### PR TITLE
ReadOnlyLockedContext support in RelationalDAO

### PR DESCRIPTION
**Pull request by AshayKing**

### CODE REVIEW

**Comment by santanusinha** [comment]
Please rebase with upstream

File: src/main/java/io/appform/dropwizard/sharding/dao/RelationalDao.java [hunk 1]
```diff
@@ -503,4 +546,193 @@ protected Field getKeyField() {
         return this.keyField;
     }
 
+    public ReadOnlyContext<T> readOnlyExecutor(final String parentKey,
+                                               final Object key) {
+        return readOnlyExecutor(parentKey, key, null);
```
Comment (r0goyal) [inline src/main/java/io/appform/dropwizard/sharding/dao/RelationalDao.java:551]: can entity populator be `() -> false` here ?

File: src/test/java/io/appform/dropwizard/sharding/dao/RelationalReadOnlyLockedContextTest.java [hunk 2]
```diff
@@ -0,0 +1,250 @@
...
+        );
+
+        val dataList = companyRelationalDao.readOnlyExecutor(parentKey, criteria, 0, 4)
+                .readAugmentParent(departmentRelationalDao, null, assosicationSpec, 0, Integer.MAX_VALUE, Company::setDepartments)
+                .readAugmentParent(ceoRelationalDao, null, assosicationSpec, 0, Integer.MAX_VALUE, (parent, childList) -> {
+                    parent.setCeo(childList.stream().findAny().orElse(null));
+                })
+                .execute()
+                .orElse(new ArrayList<>());
+        System.out.println(new ObjectMapper().writeValueAsString(dataList));
```
Comment (r0goyal) [inline src/test/java/io/appform/dropwizard/sharding/dao/RelationalReadOnlyLockedContextTest.java:160]: sout does not help in assertions. Add relevant assertions here

File: src/test/java/io/appform/dropwizard/sharding/dao/RelationalReadOnlyLockedContextTest.java [hunk 3]
```diff
@@ -0,0 +1,250 @@
...
+                .add(Restrictions.eq("companyUsageId", companyToRetrieve));
+
+        val deptCriteria = DetachedCriteria.forClass(Department.class)
+                .add(Restrictions.eq("companyExtId", companyToRetrieve));
+
+        val dataList2 = companyRelationalDao.readOnlyExecutor(parentKey, companyCriteria, 0, 4)
+                .readAugmentParent(departmentRelationalDao, deptCriteria, null, 0, Integer.MAX_VALUE, Company::setDepartments)
+                .execute()
+                .orElse(new ArrayList<>());
+        System.out.println(new ObjectMapper().writeValueAsString(dataList2));
```
Comment (r0goyal) [inline src/test/java/io/appform/dropwizard/sharding/dao/RelationalReadOnlyLockedContextTest.java:176]: Add relevant assertions here

File: src/test/java/io/appform/dropwizard/sharding/dao/RelationalReadOnlyLockedContextTest.java (hunk 2 above)
Comment (AshayKing) [reply src/test/java/io/appform/dropwizard/sharding/dao/RelationalReadOnlyLockedContextTest.java]: done

File: src/test/java/io/appform/dropwizard/sharding/dao/RelationalReadOnlyLockedContextTest.java (hunk 3 above)
Comment (AshayKing) [reply src/test/java/io/appform/dropwizard/sharding/dao/RelationalReadOnlyLockedContextTest.java:196]: done

**Pull request merged**